src/tools/makeTestfile.pl

```diff
@@ -154,9 +154,19 @@ BEGIN
     };
 
     alarm \$timeout;
-    waitpid \$pid, 0;
-    alarm 0;
-    exit \$? >> 8;
+    while (1) {
+        waitpid \$pid, 0;
+        if (WIFEXITED(\$?)) {
+            # normal exit
+            alarm 0;
+            exit WEXITSTATUS(\$?);
+        } elif (WIFSIGNALED(\$?)) {
```

Comment:
You need to load Perl's POSIX module to be able to use those macros, add the line shown below just above the call to `fork()` in line 143 above.
```perl
use POSIX qw(WEXITSTATUS WIFEXITED WIFSIGNALED);
```
I'm pretty sure this will fix all the CI failures.

Reply:
I was also missing elif -> elsif